zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/JobControllerTest.java

```diff
@@ -347,6 +350,130 @@ void shouldCompleteJobWithResultDeniedTrue() {
 
+  @Test
+  void shouldCompleteJobWithResultWithCorrections() {
+    // given
+    when(jobServices.completeJob(anyLong(), any(), any()))
+        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));
+
+    final var request =
+        """
+          {
+            "result": {
+              "denied": false,
+              "corrections": {
+                "assignee": "Test",
+                "dueDate": "2025-05-23T01:02:03+01:00",
+                "followUpDate": "2025-05-25T01:02:03+01:00",
+                "candidateUsersList": ["UserA", "UserB"],
+                "candidateGroupsList": ["GroupA", "GroupB"],
+                "priority": 20
+              }
+            }
+          }
+        """;
+
+    // when/then
+    webClient
+        .post()
+        .uri(JOBS_BASE_URL + "/1/completion")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    final ArgumentCaptor<JobResult> jobResultArgumentCaptor =
+        ArgumentCaptor.forClass(JobResult.class);
+    Mockito.verify(jobServices)
+        .completeJob(eq(1L), eq(Map.of()), jobResultArgumentCaptor.capture());
+
+    Assertions.assertEquals(
+        "Test", jobResultArgumentCaptor.getValue().getCorrections().getAssignee());
+    Assertions.assertEquals(
+        "2025-05-23T01:02:03+01:00",
+        jobResultArgumentCaptor.getValue().getCorrections().getDueDate());
+    Assertions.assertEquals(
+        "2025-05-25T01:02:03+01:00",
+        jobResultArgumentCaptor.getValue().getCorrections().getFollowUpDate());
+    Assertions.assertEquals(
+        List.of("UserA", "UserB"),
+        jobResultArgumentCaptor.getValue().getCorrections().getCandidateUsers());
+    Assertions.assertEquals(
+        List.of("GroupA", "GroupB"),
+        jobResultArgumentCaptor.getValue().getCorrections().getCandidateGroups());
+    Assertions.assertEquals(20, jobResultArgumentCaptor.getValue().getCorrections().getPriority());
+
+    Assertions.assertEquals(
+        List.of(
+            "assignee",
+            UserTaskRecord.DUE_DATE,
+            UserTaskRecord.FOLLOW_UP_DATE,
+            UserTaskRecord.CANDIDATE_USERS,
+            UserTaskRecord.CANDIDATE_GROUPS,
+            UserTaskRecord.PRIORITY),
+        jobResultArgumentCaptor.getValue().getCorrectedAttributes());
+  }
+
+  @Test
+  void shouldCompleteJobWithResultWithCorrectionsPartiallySet() {
+    // given
+    when(jobServices.completeJob(anyLong(), any(), any()))
+        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));
+
+    final var request =
+        """
+          {
+            "result": {
+              "denied": false,
+              "corrections": {
+                "assignee": "Test",
+                "candidateUsersList": ["UserA", "UserB"],
+                "candidateGroupsList": ["GroupA", "GroupB"],
+                "priority": 20
+              }
```

Comment:
🔧 To emphasize that only explicitly provided values are considered corrected, you could use `null` for some fields in the `corrections` JSON.
This would highlight how Zeebe-rest-gateway handles fields that are not updated.

For example, the test could be modified like this:

```java
  @Test
  void shouldCompleteJobWithResultWithCorrectionsPartiallySet() {
    // given
    when(jobServices.completeJob(anyLong(), any(), any()))
        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));

    final var request =
        """
          {
            "result": {
              "denied": false,
              "corrections": {
                "assignee": null,
                "dueDate": "2025-05-23T01:02:03+01:00",
                "candidateGroupsList": ["GroupA", "GroupB"],
                "priority": null
              }
            }
          }
        """;

    // when/then
    webClient
        .post()
        .uri(JOBS_BASE_URL + "/1/completion")
        .accept(MediaType.APPLICATION_JSON)
        .contentType(MediaType.APPLICATION_JSON)
        .bodyValue(request)
        .exchange()
        .expectStatus()
        .isNoContent();

    final var jobResultArgumentCaptor = ArgumentCaptor.forClass(JobResult.class);
    Mockito.verify(jobServices)
        .completeJob(eq(1L), eq(Map.of()), jobResultArgumentCaptor.capture());

    assertThat(jobResultArgumentCaptor.getValue().getCorrections())
        .isEqualTo(
            new JobResultCorrections()
                .setDueDate("2025-05-23T01:02:03+01:00")
                .setCandidateGroups(List.of("GroupA", "GroupB"))
                // The remaining fields have their default non-null values,
                // as they weren't corrected
                .setAssignee("")
                .setFollowUpDate("")
                .setCandidateUsers(List.of())
                .setPriority(-1));

    assertThat(jobResultArgumentCaptor.getValue().getCorrectedAttributes())
        .containsExactly(UserTaskRecord.DUE_DATE, UserTaskRecord.CANDIDATE_GROUPS);
  }
```

This would make it clear which fields were corrected and which were not.
Let me know what do you think about it! 😃

Reply:
I think it might even be a separate test.